crates/types/src/v1alpha1/wasmcloud_host_config.rs

```diff
@@ -187,6 +189,36 @@ fn default_nats_leafnode_port() -> u16 {
     7422
 }
 
+#[derive(Deserialize, Serialize, Clone, Debug, JsonSchema)]
+pub struct ConfigMapSource {
+    /// Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+    pub name: String,
+
+    /// Specify whether the ConfigMap must be defined
+    pub optional: Option<bool>,
+}
+
+#[derive(Deserialize, Serialize, Clone, Debug, JsonSchema)]
+pub struct SecretSource {
```

Comment:
Is there a reason you didn't just use the upstream `SecretVolumeSource` for this? It has some additional fields, but they're all optional

Reply:
you're 🎯 . upstream was the goal https://gist.github.com/lxfontes/15c58f063ec260da816ff5ffc09f6cbf

The ⚖️  is that `SecretVolumeSource` has `secretName` while `ConfigMapVolumeSource` has `name`.
I'd like to reuse the upstreams too. @protochron @joonas what do you think?

Reply:
I think adopting the `<thing>VolumeSource` for this makes sense, we can cover the differences in terms of what configuration options needs to be provided with each of the available `VolumeSource`s in a more thorough example and probably as part of the [wasmCloud docs](https://wasmcloud.com/docs/) site.